src/braket/default_simulator/simulator.py

```diff
@@ -53,16 +59,79 @@
 
 
 class BaseLocalSimulator(BraketSimulator):
-    @property
-    @abstractmethod
-    def device_action_type(self) -> DeviceActionType:
-        """DeviceActionType"""
-
-    @abstractmethod
+    @singledispatchmethod
     def run(
```

Comment:
Rather than making `run` a `singledispatchmethod`, create a private `_run` method for dispatching instead; public functions marked with `singledispatch` end up producing really ugly docs (you end up with one extra run method, but with blank documentation, for each implementation)

Reply:
Because of the differing signatures for the run methods of each IR, I kept the signature of the top-level run method general, with a detailed doc string going more in depth about the specific parameters used. I also made the ir-specific run methods public so that the detailed documentation for each ir type would be generated.